### DISCUSSION TITLE
How to use Diagram Creation in Tree Description

### DISCUSSION THREAD

**Discussion by hoajie** [eclipse-sirius/sirius-desktop#483] [Q&A] [answered]
Dear Sirius Community,

I hope this message finds you well.

I am currently using Sirius for modeling and have encountered some difficulties in understanding how to create a new diagram from a Tree Description.
I mean right clicking in Tree Description shows New Representation Creation, and inside that there is Diagram Creation. I'm curious if this allows the creation of new Diagrams? But I don't know how to use this.
 I’ve gone through the documentation, but I’m still unsure about the specific steps required to achieve this.

Could someone kindly explain or guide me on how to set up Diagram Creation from a Tree Description? Any examples or references to relevant documentation would also be greatly appreciated.

Thank you in advance for your assistance, and I look forward to any insights you can provide.

Best regards,
![Snipaste_2024-10-22_17-27-16](https://github.com/user-attachments/assets/4e9c506a-2d15-4429-83f6-f59860eb2517)
![Snipaste_2024-10-22_17-27-37](https://github.com/user-attachments/assets/744cdb7e-ca9e-4fce-a7c3-8d3f6419b304)
![Snipaste_2024-10-22_17-27-56](https://github.com/user-attachments/assets/48431977-bdf4-4dcf-8013-f1c8f8bdf54a)

**Comment by lredor** [comment, marked as answer]
Hi,
I've made a sample (base on [this starting point](https://github.com/eclipse-sirius/sirius-desktop/tree/master/plugins/org.eclipse.sirius.tests.swtbot/data/unit/tree/navigation)) and I've added a "Diagram creation" tool available on packages.

Steps to reproduce:
- Import the project (made with ODCE 11.8.0) [TreeSampleProject.zip](https://github.com/user-attachments/files/17474681/TreeSampleProject.zip)
- Open the "new Tree" representation
- Right-click on "Package 3" and launch the action "New/Diagram Creation on Package"
- Expected: A new diagram is created and opened.

I hope this can help you.

Regards,

Laurent

**Reply by hoajie** [reply]
Dear Laurent Redor,

Thank you so much for your swift and well-structured response! I really appreciate the clarity of the steps you provided. I will definitely follow your suggestions and give it a try.

Once again, thank you for your support and for taking the time to help me. Your assistance is greatly appreciated.

Best regards,
